src/api/Feed.js

```diff
@@ -628,8 +707,41 @@ class Feed extends Base {
                     return;
                 }
 
-                const parsedFeedItems = getParsedFileActivitiesResponse(response);
-                handleFeedItems(parsedFeedItems);
+                Promise.all([
+                    versionsPromise,
+                    currentVersionPromise,
+                    commentsPromise(),
+                    tasksPromise,
+                    appActivityPromise,
+                    annotationsPromise,
+                ]).then(
+                    ([
+                        versions: ?FileVersions,
+                        currentVersion: ?BoxItemVersion,
+                        comments,
+                        tasks,
+                        appActivity,
+                        annotations,
+                    ]) => {
+                        // Get parity data between V2 and UAA
+                        const v2ShadowItems = { versions, currentVersion, comments, tasks, appActivity, annotations };
+                        const versionsWithCurrent = currentVersion
+                            ? this.versionsAPI.addCurrentVersion(currentVersion, versions, this.file)
+                            : undefined;
+                        const sortedFeedItems = sortFeedItems(
+                            versionsWithCurrent,
+                            comments,
+                            tasks,
+                            appActivity,
+                            annotations,
+                        );
+                        const responseParity = getResponseParity(response, v2ShadowItems);
+
+                        const parsedFeedItems = getParsedFileActivitiesResponse(response);
+                        const parsedDataParity = { v2: sortedFeedItems, uaa: parsedFeedItems };
+                        handleFeedItems(parsedFeedItems, { responseParity, parsedDataParity });
```

Comment:
So we're not going to handle the feed items until all of the promises resolve? Will this double the time for the results to be shown to the user. Should the parity check not be an asynchronous process that will log an error message if the results from UAA are different from the results coming from the individual api calls?

Reply:
That makes sense, I updated the PR so that the UI is not blocked by both responses. Now the v2 calls will be async and will not be blocking and will be used only for logging.